.github/workflows/sbom.yml

```
@@ -0,0 +1,47 @@
+name: SBOM generation
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  sbom_gen:
+    name: Generate SBOM
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+          cache: 'pip'
+          cache-dependency-path: '**/requirements.txt'
+      - name: Get date
+        id: get-date
+        run: |
+          echo "::set-output name=date::$(/bin/date -u "+%Y%m%d")"
+      - name: Get cached database
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/cve-bin-tool
+          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+      - name: Install dependencies and cve-bin-tool
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade setuptools
+          python -m pip install --upgrade wheel
+          python -m pip install --upgrade pytest
+          python -m pip install --upgrade sbom4python
+          pip install . -r doc/requirements.txt
+      - name: Generate SBOM for cve-bin-tool
+        run: |
+          sbom4python --module cve-bin-tool --output cve_bin_tool.spdx
+      - name: Compare SBOM for cve-bin-tool
+        # This would fail due to time/date of SBOM generation in SBOM header
+        # Therefore ignore first 10 lines of file in comparison which is SBOM header
+        run: |
+          /bin/tail -n +10 sbom/cve_bin_tool.spdx > orig
+          /bin/tail -n +10 cve_bin_tool.spdx > new
```

```suggestion
          /bin/tail -n +10 sbom/cve-bin-tool.spdx > orig
          /bin/tail -n +10 cve-bin-tool.spdx > new
```

Love the tail solution!  Looks like you changed between _ and - somewhere so the files don't match when this runs:

```
Run /bin/tail -n +10 sbom/cve_bin_tool.spdx > orig
/bin/tail: cannot open 'sbom/cve_bin_tool.spdx' for reading: No such file or directory
```